misc/testerina/modules/testerina-runtime/src/main/java/org/ballerinalang/test/runtime/entity/MockFunctionReplaceVisitor.java

```diff
@@ -0,0 +1,149 @@
+import static org.objectweb.asm.Opcodes.INVOKESTATIC;
+import static org.objectweb.asm.Opcodes.LLOAD;
+import static org.objectweb.asm.Opcodes.LRETURN;
+import static org.objectweb.asm.Opcodes.RETURN;
+
+/**
+ * Remove existing method body and replace it with a method call.
+ *
+ * @since 2201.1.0
+ */
+public class MockFunctionReplaceVisitor extends ClassVisitor {
+
+    private final String methodName;
+    private final String methodDesc;
+    private final Method mockFunc;
+
+    public MockFunctionReplaceVisitor(int api, ClassWriter cw, String name, String methodDescriptor, Method mockFunc) {
+        super(api, cw);
+        this.methodName = name;
+        this.methodDesc = methodDescriptor;
+        this.mockFunc = mockFunc;
+    }
+
+    @Override
+    public MethodVisitor visitMethod(int access, String name, String desc, String signature, String[] exceptions) {
+
+        if (!name.equals(methodName) || !desc.equals(methodDesc)) {
+            // reproduce the methods where no changes needed
+            return super.visitMethod(access, name, desc, signature, exceptions);
+        }
+
+        return new MethodReplaceAdapter(
+                super.visitMethod(access, name, desc, signature, exceptions), mockFunc);
+    }
+
+
+    private static class MethodReplaceAdapter extends MethodVisitor {
+
+        private final Method mockFunc;
+
+        MethodReplaceAdapter(MethodVisitor mv, Method mockFunc) {
+            super(Opcodes.ASM7, mv);
+            this.mockFunc = mockFunc;
+        }
+
+        @Override
+        public void visitCode() {
+
+            Label label0 = new Label();
+            super.visitLabel(label0);
+            super.visitMethodInsn(INVOKESTATIC, TESTERINA_UTILS, IS_CALL_ORIGINAL, "()Z", false);
+            Label label1 = new Label();
+            super.visitJumpInsn(IFNE, label1);
+            Label label2 = new Label();
+            super.visitLabel(label2);
+
+            Class<?>[] parameterTypes = mockFunc.getParameterTypes();
+            int paramOffset = 0;
+            for (Class<?> parameterType : parameterTypes) {
+                generateLoadInstruction(parameterType, paramOffset);
+                if (parameterType == Long.TYPE || parameterType == Double.TYPE) {
+                    paramOffset += 2;
+                } else {
+                    paramOffset++;
+                }
+            }
+
+            String mockFuncClassName = mockFunc.getDeclaringClass().getName().replace(".", "/");
+            super.visitMethodInsn(INVOKESTATIC, mockFuncClassName, mockFunc.getName(),
+                    Type.getMethodDescriptor(mockFunc), false);
+
+            generateReturnInstruction(mockFunc.getReturnType());
+            super.visitLabel(label1);
+            super.visitCode();
+        }
+
+        private void generateLoadInstruction(Class<?> type, int index) {
+            if (type.isPrimitive()) {
+                if (type == Integer.TYPE || type == Boolean.TYPE) {
```

Comment:
Shall we use switch case statements in both `generateLoadInstruction` and `generateReturnInstruction` methods

Reply:
Since we can't use switch statement to switch between `Class` objects without defining tags for them or defining `enum`. And since the number of cases is small here, we can go with `if-else` statements.